src/hooks/useCollectionsData.ts

```diff
@@ -0,0 +1,61 @@
+import { StorageKey, u32 } from '@polkadot/types';
+import { PalletUniquesCollectionMetadata } from '@polkadot/types/lookup';
+import { AccountId32 } from '@polkadot/types/interfaces';
+import { useCallback, useEffect, useState } from 'react';
+
+import { CollectionMetadata } from '../constants';
+import { useAccounts } from '../Contexts';
+
+const useCollectionsData = () => {
+  const { api, activeAccount } = useAccounts();
+  const [ownedCollectionIds, setOwnedCollectionIds] = useState<string[] | null>(null);
+  const [collectionsMetadata, setCollectionsMetadata] = useState<CollectionMetadata[] | null>(null);
+
+  const getAccountCollectionsIds = useCallback(async () => {
+    if (api && activeAccount) {
+      const results: StorageKey<[AccountId32, u32]>[] = await api.query.uniques.classAccount.keys(activeAccount.address);
+      const collectionIds = results
+        .map(({ args: [, collectionId] }) => collectionId)
+        .sort((a, b) => a.cmp(b))
+        .map((collectionId) => collectionId.toString());
+
+      setOwnedCollectionIds(collectionIds);
+    }
+
+    return [];
+  }, [api, activeAccount]);
+
+  const getCollectionsData = useCallback(async () => {
+    if (api && ownedCollectionIds) {
+      let metadata: CollectionMetadata[] = [];
+      const rawMetadata = await api.query.uniques.classMetadataOf.multi(ownedCollectionIds);
+
+      if (Array.isArray(rawMetadata) && rawMetadata.length > 0) {
+        const fetchCalls = rawMetadata.map((metadata) => {
+          const primitiveMetadata = metadata.toPrimitive() as unknown as PalletUniquesCollectionMetadata;
+          return fetch(`https://ipfs.io/ipfs/${primitiveMetadata.data}`);
```

Comment:
Let's make the gateway configurable since we might use a custom gateway other than https://ipfs.io/ipfs or even a round-robin set with fallbacks in the future.

Reply:
Yeah, I'll add a config 👍